lib/bigquery/job.js

```diff
@@ -156,35 +188,9 @@ nodeutil.inherits(Job, ServiceObject);
  * @example
  * job.cancel(function(err, apiResponse) {
  *   // Check to see if the job completes successfully.
- *   onJobComplete(function(err) {
- *     if (!err) {
- *       // Job cancelled successfully.
- *     }
- *   });
+ *   job.on('error', function(err) {});
+ *   job.on('complete', function(metadata) {});
```

Comment:
Returning an emitter is an awesome idea. I will open a new issue for that, since implementing it will also affect GCE at the same time.